src/scenes/Engagement/Files/useUploadEngagementFiles.ts

```diff
@@ -0,0 +1,54 @@
+import { GQLOperations } from '../../../api';
+import {
+  HandleUploadCompletedFunction,
+  UploadFilesConsumerFunction,
+  UploadFilesConsumerInput,
+  useUploadFiles,
+} from '../../../components/files/hooks';
+import {
+  useUploadInternshipEngagementGrowthPlanMutation,
+  useUploadLanguageEngagementPnpMutation,
+} from './UploadEngagementFiles.generated';
+
+type EngagementType = 'language' | 'internship';
+
+export const useUploadEngagementFile = (
+  engagementType: EngagementType
+): UploadFilesConsumerFunction => {
+  const uploadFiles = useUploadFiles();
+
+  const [uploadPnp] = useUploadLanguageEngagementPnpMutation();
+  const [uploadGrowthPlan] = useUploadInternshipEngagementGrowthPlanMutation();
+
+  const uploadFile =
+    engagementType === 'language' ? uploadPnp : uploadGrowthPlan;
+
+  const handleUploadCompleted: HandleUploadCompletedFunction = async ({
+    uploadId,
+    name,
+    parentId: id,
+    action,
+  }) => {
+    await uploadFile({
+      variables: {
+        id,
+        pnp: { uploadId, name },
+        growthPlan: { uploadId, name },
+      },
+      refetchQueries: [
+        action === 'version'
+          ? GQLOperations.Query.FileVersions
+          : GQLOperations.Query.Engagement,
+      ],
+    });
+  };
+
+  const uploadEngagementFile = ({
+    action,
+    files,
+    parentId,
+  }: UploadFilesConsumerInput) =>
+    uploadFiles({ action, files, handleUploadCompleted, parentId });
```

Comment:
I'm seeing a few places where we accept/pass around a list of files. I don't think it makes sense to upload more than one file at a time for these defined files, regardless of first or not. Do we limit this in some spot? If it's easier to pass around the list and then filter down to a single file in one spot then that makes sense too.

Reply:
We can limit it at the Dropzone, which I'm pretty sure we do for Engagement files. But even if you limit to one it always passes that one around in an `Array`

Reply:
Well, I'll be, we actually weren't limiting it. Fixed now